Source/JavaScriptCore/dfg/DFGIntegerRangeOptimizationPhase.cpp

```diff
@@ -1184,84 +1184,98 @@ class IntegerRangeOptimizationPhase : public Phase {
                 Node* terminal = block->terminal();
                 bool alreadyMerged = false;
                 if (terminal->op() == Branch) {
-                    Relationship relationshipForTrue;
+                    Vector<Relationship, 1> relationshipForTrue;
+                    Vector<Relationship, 1> relationshipForFalse;
```

Comment:
Could you please leave a comment `All facts in both relationshipForTrue and relationshipForFalse are ALWAYS combined with AND.` here?

Reply:
The opposite is true in this case. They must all hold independently, and it is always correct to remove a relationship. That is why I can't also add the inverse cases (i.e, branch(BitAnd(a, b)) gives a and b on the true branch, but nothing on the false branch).